src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx

```diff
@@ -0,0 +1,138 @@
+> = {
+  staffName: {
+    header: 'Staff Name',
+    render: (assignment) => assignment.user.name,
+  },
+  role: {
+    header: 'Role',
+    render: (assignment) => assignment.staffAssignmentType,
+  },
+  assignmentDate: {
+    header: 'Assignment Date',
+    render: (assignment) => parseAndFormatDate(assignment.assignedAt),
+  },
+  assignmentDateRange: {
+    header: 'Assignment Dates',
+    render: (assignment) =>
+      parseAndFormatDateRange(assignment.assignedAt, assignment.unassignedAt),
+  },
+  action: {
+    header: 'Action',
+    render: (assignment) => (
+      <UnassignStaffButton staffAssignmentId={assignment.id} />
+    ),
+  },
+};
+
+interface EditStaffAssignmentDialogProps {
+  household: HouseholdWithStaffAssignmentsFragment;
+  open: boolean;
+  onClose: VoidFunction;
+}
+
+const EditStaffAssignmentDialog: React.FC<EditStaffAssignmentDialogProps> = ({
+  household,
+  open,
+  onClose,
+}) => {
+  const isTiny = useIsMobile('sm');
+
+  const headOfHousehold = useMemo(() => {
+    return household.householdClients.find(
+      (c) => c.relationshipToHoH === RelationshipToHoH.SelfHeadOfHousehold
+    );
+  }, [household.householdClients]);
+
+  const { enrollment } = useEnrollmentDashboardContext();
+
+  if (!enrollment) return;
+  if (!headOfHousehold) return;
+
+  return (
+    <CommonDialog fullWidth open={open} onClose={onClose}>
+      <DialogTitle>Manage Staff Assignments</DialogTitle>
+      <DialogContent>
+        <Stack gap={2} sx={{ mt: 2 }}>
+          <Stack direction={isTiny ? 'column' : 'row'} gap={isTiny ? 2 : 4}>
+            <CommonLabeledTextBlock title='Head of Household'>
+              {clientBriefName(headOfHousehold.client)}
+            </CommonLabeledTextBlock>
+            <CommonLabeledTextBlock title='Project'>
+              {enrollment.project.projectName}
+            </CommonLabeledTextBlock>
+            <CommonLabeledTextBlock title='Enrollment Period'>
+              <EnrollmentDateRangeWithStatus enrollment={enrollment} />
+            </CommonLabeledTextBlock>
+          </Stack>
+          <NewStaffAssignmentForm
+            householdId={household.id}
+            projectId={enrollment.project.id}
+          />
+          <Typography
+            id='current-staff-assignments'
+            sx={{ mt: 1 }}
+            variant='h5'
+            component='h3'
+          >
+            Current Staff Assignments
+          </Typography>
+          <Card>
+            <GenericTable<StaffAssignmentDetailsFragment>
+              rows={household.staffAssignments?.nodes || []}
+              columns={[
+                STAFF_ASSIGNMENT_COLUMNS.staffName,
+                STAFF_ASSIGNMENT_COLUMNS.role,
+                STAFF_ASSIGNMENT_COLUMNS.assignmentDate,
+                STAFF_ASSIGNMENT_COLUMNS.action,
+              ]}
+              noData={'No Staff Assigned'}
+              tableProps={{ 'aria-label': 'Current staff assignments' }}
+            />
```

Comment:
This is fine as-is but maybe it would be a nicer to extract into another component, StaffAssignmentTable, along with the STAFF_ASSIGNMENT_COLUMNS constant.

Reply:
Good plan 👍